internal/clients/sts/sts_client.go

```diff
@@ -0,0 +1,36 @@
+package sts
+
+import (
+	"context"
+
+	"github.com/RHEnVision/provisioning-backend/internal/config"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	c "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"github.com/aws/aws-sdk-go-v2/service/sts/types"
+)
+
+func AssumeRole(arn string) (*types.Credentials, error) {
+	stsClient, err := CreateSTSClient()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
The function should have client as a reciever, therefore it should not be creating any client.